### PR TITLE
Feature/guilds responsiveness

### PR DESCRIPTION
**Pull request by luchux**
This works adds responsiveness to work done on Guilds. 
- introduces react-device-detection package, to use isDesktop, isMobile conditional rendering. 
- refactored the sidebar to DesktopSidebar and MobileSidebar to be use in each case. 
- added media queries to proposal list of guild, header, and other components. 
screencapture-localhost-3000-2021-11-25-12_50_54
screencapture-localhost-3000-2021-11-25-12_49_29

### CODE REVIEW

File: src/components/Guilds/Sidebar/SidebarMobile.tsx [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+*/
+const UserInfoPanel = styled(Box)`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  padding: 1.2rem;
+  border-top: 1px solid #000;
+`;
+
+const UserInfoDetail = styled(Box)`
```
Comment (luchux) [inline src/components/Guilds/Sidebar/SidebarMobile.tsx]: so, here SidebarMobile and DesktopMobile share some common styles (the ones that are the same for mobile and desktop), should we separate them in a common.styled.ts file ?

File: src/components/Guilds/Sidebar/SidebarMobile.tsx [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+              <DaoIcon src={dxIcon} alt={'DXdao Logo'} />
+              <DaoTitle as="h1">DXdao</DaoTitle>
+            </DaoBrand>
+
+            <DaoMemberCount>464 Members</DaoMemberCount>
+          </DaoInfoRow>
+          {!showExtendedHeader && <Button>Create Proposal</Button>}
+        </DaoInfoPanel>
+
+        {showExtendedHeader && (
```
Comment (luchux) [inline src/components/Guilds/Sidebar/SidebarMobile.tsx]: The work of Madush and the figma designs were different in this section. I guess was after made work, I will talk to Madush in the standup to see if we keep this, if we write the new based on whats in the figma, and if we keep this sidebars separated or we share among both with conditional rendering.

**Comment by hhamud** [comment]
Format check has failed. run the formatting tool when you can

**Comment by rossneilson** [comment]
Im seeing a few weird behaviours different to the figma and screenshot in the PR description
FIX: need to reload

Screenshot 2021-11-25 at 15 08 04

File: src/components/Guilds/Sidebar/SidebarMobile.tsx (hunk 2 above)
Comment (mprasanjith) [reply src/components/Guilds/Sidebar/SidebarMobile.tsx]: I think you got this sorted already :) Basically, the Figma designs are the latest, this current code was written earlier.

**Review by rossneilson — DISMISSED** [review]

**Review by rossneilson — APPROVED** [review]

**Pull request merged**